clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -37,9 +41,42 @@
     public static final String FS_LAKEFS_SECRET_KEY = "fs.lakefs.secret.key";
     private static final String BASIC_AUTH = "basic_auth";
 
+    private Configuration conf;
     private URI uri;
     private Path workingDirectory = new Path("/");
-    private io.lakefs.clients.api.ApiClient apiClient;
+    private ApiClient apiClient;
+
+    private static class ParsedPath {
+	ParsedPath(String repo, String ref, String path) {
+	    this.repo = repo;
+	    this.ref = ref;
+	    this.path = path;
+	}
+	public String repo;
+	public String ref;
+	public String path;
+    }
+
+    private ParsedPath parsePath(Path path) {
+	if (!path.isAbsolute()) path = new Path(workingDirectory, path);
+
+	URI pathUri = path.toUri();
+	String[] parts = pathUri.getPath().split("/", 3);
+	if (!parts[0].equals("")) {
+	    throw new RuntimeException("path does not start with / (" + String.join(" [/] ", parts) + ")");
+	}
+	return new ParsedPath(pathUri.getAuthority(), parts[1], parts[2]);
```

Comment:
used the getHost for this one as we don't use the port for the repository - not sure which one is the better